src/ansys/systemcoupling/core/charts/chart_datatypes.py

```diff
@@ -0,0 +1,106 @@
+# Copyright (C) 2023 - 2024 ANSYS, Inc. and/or its affiliates.
+# SPDX-License-Identifier: MIT
+#
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from dataclasses import dataclass, field
+from enum import Enum
+from typing import Optional
+
+# Common intermediate metadata and data types for chart series information.
+# All series are assumed to belong to an interface and a data transfer within the interface
+
+# Idea is that data should be processed into this format from whatever raw source (CSV file,
+# streamed data, etc.) and the actual charts will be built from this.
+
+
+class SeriesType(Enum):
+    CONVERGENCE = 1
+    SUM = 2
+    WEIGHTED_AVERAGE = 3
+
+
+@dataclass
+class TransferSeriesInfo:
+    """Information about chart series data associated with a data transfer.
+
+    The `data_index` fields are used by the data source processor to associate
+    this information (likely obtained from heading or other metadata) with the
+    correct data series. It indexes into the full list of series associated
+    with a given interface.
+
+    `line_suffixes` is either empty or contains the suffixes for
+    component series of various types (real/imag and or x/y/z).
+    It is assumed that such components can be found via the index
+    information - i.e. that they will be contiguous from the
```

Comment:
```suggestion
    It is assumed that such components can be found using the index
    information. For example, they are contiguous from the
```
Google dev doc style guide says not to use "via" in documentation. Also use present tense and avoid Latin phrases.

Reply:
Fair enough but Google's advice seem like a weird restriction! To me, "via" carries more of the sense of the index getting you from one place to another.

If we need to expand "i.e.", shouldn't it be "that is"? I wasn't giving an example, I was expanding on the information.

Reply:
The above comments are probably moot as I am making local changes that extend and reword the docstrings of this file.

Reply:
Doc strings in this module have been overhauled and are very different from what was reviewed so will mark this conversation as resolved.